### PR TITLE
url subcommand returns http even if https is configured

### PR DESCRIPTION
**Pull request by cu12**
I recently upgraded to the latest version and noticed that the behaviour of `dokku url APP` has been changed and returns `http` scheme instead of `https`. This unfortunately breaks the deployment for some of my applications.

I know about the `dokku urls` command, although I was wondering if this change was really intended as I couldn't find anything in release notes. Also since `http` really just redirects to `https` if certs are configured it's really not the correct address to display.

If you think this should go in, I'll add a test as well.

### CODE REVIEW

**Comment by josegonzalez** [comment]
Definitely needs a test.

**Comment by michaelshobbs** [comment]
This was most likely a side-effect of the proxy port mapping stuff. Might be a better solution to grab all https urls if a cert is configured and pick the first or last of those instead of just inverting the logic of `dokku url APP`. My 2 cents...

**Comment by josegonzalez** [comment]
@cu12 Mind updating the PR with @michaelshobbs comment?

**Comment by josegonzalez** [comment]
Tests fail because dokku.me doesn't have https. Not sure what the best thing to do here is... @michaelshobbs ?

File: plugins/common/functions [hunk 1]
```diff
@@ -976,7 +977,7 @@ get_app_urls() {
       local port_map app_vhost
       local app_vhosts=$(get_app_domains "$APP")
       for port_map in $DOKKU_PROXY_PORT_MAP; do
-        local scheme="$(awk -F ':' '{ print $1 }' <<< "$port_map")"
+      local scheme="$(awk -F ':' '{ print $1 }' <<< "$port_map")"
```
Comment (michaelshobbs) [inline plugins/common/functions]: need to indent this line

File: tests/unit/40_core_2.bats [hunk 2]
```diff
@@ -20,6 +20,15 @@ assert_urls() {
   assert_output < <(tr ' ' '\n' <<< "${urls}" | sort)
 }
 
+assert_url() {
```
Comment (michaelshobbs) [inline tests/unit/40_core_2.bats]: why do we need this when we already have`assert_urls`?

Comment (michaelshobbs) [inline tests/unit/40_core_2.bats]: nvm. i see now

**Comment by michaelshobbs** [comment]
The error is due to a syntax error. Additionally the logic of the new test doesn't make sense. specifically these lines:

```
add_domain "test.dokku.me"
assert_url "https://test.dokku.me" "https://${TEST_APP}.dokku.me"
``` 

`assert_url` calls `dokku url $APP` and will only over return the very first url. i think this patch to the PR is sufficient to test the desired behavior

```
diff --git a/plugins/common/functions b/plugins/common/functions
index 041a129..5857883 100755
--- a/plugins/common/functions
+++ b/plugins/common/functions
@@ -977,7 +977,7 @@ get_app_urls() {
       local port_map app_vhost
       local app_vhosts=$(get_app_domains "$APP")
       for port_map in $DOKKU_PROXY_PORT_MAP; do
-      local scheme="$(awk -F ':' '{ print $1 }' <<< "$port_map")"
+        local scheme="$(awk -F ':' '{ print $1 }' <<< "$port_map")"
         local listen_port="$(awk -F ':' '{ print $2 }' <<< "$port_map")"
         for app_vhost in $app_vhosts; do
           if [[ "$listen_port" != "80" ]] && [[ "$listen_port" != "443" ]]; then
diff --git a/tests/unit/40_core_2.bats b/tests/unit/40_core_2.bats
index 69f5bbe..7608171 100644
--- a/tests/unit/40_core_2.bats
+++ b/tests/unit/40_core_2.bats
@@ -26,7 +26,7 @@ assert_url() {
   echo "output: "$output
   echo "status: "$status
   echo "url: ${url}"
-  assert_output < <("${url}")
+  assert_output "${url}"
 }

 build_nginx_config() {
@@ -76,8 +76,6 @@ build_nginx_config() {
   assert_url "https://dokku.me"
   build_nginx_config
   assert_url "https://${TEST_APP}.dokku.me"
-  add_domain "test.dokku.me"
-  assert_url "https://test.dokku.me" "https://${TEST_APP}.dokku.me"
 }
```

**Comment by michaelshobbs** [comment]
@cu12 let me know if that makes sense or not

**Comment by josegonzalez** [comment]
@cu12 mind updating your pr?

**Comment by josegonzalez** [comment]
@cu12 *bump*

**Comment by josegonzalez** [comment]
@cu12 *bump*

**Comment by josegonzalez** [comment]
Closing in favor of #2848

**Pull request closed without merging**